tensorboard/plugins/histogram/tf_histogram_dashboard/histogram_core_test.ts

```diff
@@ -0,0 +1,141 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {
+  BackendHistogramBin,
+  backendToIntermediate,
+  backendToVz,
+  intermediateToD3,
+} from './histogramCore';
+
+describe('histogram core', () => {
+  describe('backendToIntermediate', () => {
+    it('handles empty case', () => {
+      expect(backendToIntermediate([0, 0, []])).toEqual({
+        wall_time: 0,
+        step: 0,
+        min: undefined,
+        max: undefined,
+        buckets: [],
+      });
+    });
+
+    it('converts backend histogram to intermediate', () => {
+      const bins: BackendHistogramBin[] = [
+        [2, 3, 1],
```

Comment:
This order is a little odd, since in real data it would presumably be `[1, 2, 2], [2, 3, 1], [3, 4, 1]` so the bucket boundaries are in ascending order?

Reply:
Fixed.